llvm/lib/AsmParser/LLLexer.cpp

```diff
@@ -262,6 +274,38 @@ void LLLexer::SkipLineComment() {
   }
 }
 
+/// SkipCComment - This skips C-style /**/ comments.  The only difference from C
+/// is that we allow nesting.
```

Comment:
Is there a good reason to support this / are there other languages out there that support nesting C-style comments? I don't think it's a good idea to deviate from C here unless there's a very good reason to.

Reply:
This concept appears to have been copied from TGLexer.cpp

Reply:
Interesting. Dates back to af282ae26e784d42b15ae3aff887f4f3769eddec, July 2003.

Reply:
Dropped nested comment support. I suspect at that point, TableGen did not support preprocessor macros, so made sense to support nested C style comments to enable commenting out code (?)